lib/src/main/java/com/tubitv/media/fsm/state_machine/FsmPlayer.java

```diff
@@ -192,6 +192,17 @@ public State getCurrentState() {
         return currentState;
     }
 
+    @Override
+    public void restart() {
+        getController().getContentPlayer().stop();
+        getController().getContentPlayer().setPlayWhenReady(false);
+        currentState = null;
+        getController().clearMovieResumeInfo();
+
+        getController().getContentPlayer().prepare(movieMedia.getMediaSource(),true,true);
+        transit(Input.INITIALIZE);
```

Comment:
Is the point of this block to restart the video (at it's current position) after leaving the app and returning?
Why is this now needed?

Reply:
This is the restart method in state Machine to prepare restart of a new video. We need to re-set the state machine to initialize state, so that all Exoplayer related resources can be used again. You can think about this method as open a brand new "Activity" without actually relaunch the Activity